CMakeLists.txt

```diff
@@ -1,28 +1,23 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(filters)
 
-##############################################################################
-# Find dependencies
-##############################################################################
-
-find_package(catkin REQUIRED COMPONENTS pluginlib roslib roscpp rosconsole)
-find_package(Boost REQUIRED)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-include_directories(
-    include
-    ${catkin_INCLUDE_DIRS}
-)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
 ##############################################################################
-# Define package
+# Find dependencies
 ##############################################################################
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES mean params increment median transfer_function
-  CATKIN_DEPENDS roslib roscpp rosconsole pluginlib
-  DEPENDS Boost
-)
+find_package(ament_cmake REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(Boost REQUIRED)
```

Comment:
I'm not sure. I don't see a component for it, but I'm not 100% sure this lists all the component types

https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindBoost.cmake#L1201-1255

Reply:
Ah, I think this provides the answer: https://stackoverflow.com/questions/26749461/how-to-determine-the-boost-library-names-for-cmake

Since circular_buffer is a header-only, templated piece of Boost, there is no component to find.  So we just have to do this bit, and depend on libboost-dev in the package.xml

Reply:
depend on `libboost-dev` in 5056e31a40e1ec0e2c9000bcaec1f7fc1b2dd266